mosquitto/hooks/pre-start

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+# This pre-start script copies the default config if there is no config given.
+# If you want to start with new default config, you have to un- and reinstall the app.
+
+APP_DATA_DIR="$(readlink -f "$(dirname "${BASH_SOURCE[0]}")/..")"
+CONFIG_FILE="${APP_DATA_DIR}/config/mosquitto.conf"
+
+if [ -f "${CONFIG_FILE}" ]; then
+  echo "configuration file present, nothing will be changed."
+else
+  echo "configuration file NOT present, defaults will be copied."
+  cp "${CONFIG_FILE}.default" "${CONFIG_FILE}"
+  # set correct owner
+  chown umbrel:umbrel "${CONFIG_FILE}"
```

Comment:
Nice hook 🫡

Looking at the [mosquitto Dockerfile](https://github.com/eclipse-mosquitto/mosquitto/blob/master/docker/2.0/Dockerfile), we're actually stuck running the container as user:group `1883:1883` since they have hardcoded this into the image unfortunately. I think this should still work as-is though.

You can see that all data directories for mosquitto end up owned by 1883

```
:~$ ls -l ~/umbrel/app-data/mosquitto/
total 20
drwxr-xr-x 1   1883   1883   72 Nov 28 05:14 config
drwxr-xr-x 1   1883   1883    0 Nov 28 02:04 data
-rw-r--r-- 1 umbrel umbrel  574 Nov 28 04:59 docker-compose.yaml
-rw-r--r-- 1 umbrel umbrel  583 Nov 28 05:14 docker-compose.yml
drwxr-xr-x 1 umbrel umbrel   18 Nov 28 05:14 hooks
drwxr-xr-x 1   1883   1883    0 Nov 28 02:04 log
-rw-r--r-- 1 root   root     17 Nov 28 05:14 settings.yml
-rw-r--r-- 1 umbrel umbrel 1008 Nov 28 04:59 umbrel-app.yaml
-rw-r--r-- 1 umbrel umbrel 1008 Nov 28 04:59 umbrel-app.yml
drwxr-xr-x 1   1883   1883   58 Nov 28 02:04 www
```